### PR TITLE
Give XPIs to all Linux/Unix users

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-frontend/issues/2998

For user agent reference, see https://github.com/faisalman/ua-parser-js

### CODE REVIEW

**Comment by tofumatt** [comment]
Got this one too.

File: src/core/installAddon.js [hunk 1]
```diff
@@ -46,8 +46,23 @@ import {
...
+  USER_AGENT_OS_BSD_NETBSD,
+  USER_AGENT_OS_BSD_OPENBSD,
+  USER_AGENT_OS_BSD_PC,
   USER_AGENT_OS_LINUX,
+  USER_AGENT_OS_LINUX_ARCH,
+  USER_AGENT_OS_LINUX_CENTOS,
+  USER_AGENT_OS_LINUX_DEBIAN,
+  USER_AGENT_OS_LINUX_FEDORA,
+  USER_AGENT_OS_LINUX_GENTOO,
+  USER_AGENT_OS_LINUX_GNU,
```
Comment (tofumatt) [inline src/core/installAddon.js]: [But aren't *all* Linuxes GNU? Stallman would be angry.](https://www.gnu.org/gnu/linux-and-gnu.en.html) 😉

File: src/core/installAddon.js [hunk 2]
```diff
@@ -165,9 +180,26 @@ export function makeMapDispatchToProps({ WrappedComponent, src }) {
 
 const userAgentOSToPlatform = {
   [USER_AGENT_OS_ANDROID]: OS_ANDROID,
-  [USER_AGENT_OS_LINUX]: OS_LINUX,
   [USER_AGENT_OS_MAC]: OS_MAC,
   [USER_AGENT_OS_WINDOWS]: OS_WINDOWS,
+  // Not all of these are strictly Linux but giving them a Linux XPI
```
Comment (tofumatt) [inline src/core/installAddon.js]: Should we maybe file a bug to rename "Linux" to "*NIX" or something in the UI? I dunno, both options are kinda meh but I wonder which is better...

File: tests/unit/core/TestInstallAddon.js [hunk 3]
```diff
@@ -297,6 +297,20 @@ describe(__filename, () => {
         .toEqual('https://a.m.o/files/linux.xpi');
     });
 
+    it('finds a Linux Ubuntu install URL', () => {
```
Comment (tofumatt) [inline tests/unit/core/TestInstallAddon.js]: Can we have an extra one that is 100% not Linux but UNIX instead, just for completeness/clarity? I know a user was asking if we could load Linux XPIs for BSD users, so it'd be nice to have a test that spells out that intention and makes sure it stays true.

Comment (kumar303) [reply tests/unit/core/TestInstallAddon.js]: Good call. I had actually made a typo (UNIX not Unix) so it wasn't even working. I'll double check the mapping but it's probably overkill to add a test for each one.

Comment (tofumatt) [reply tests/unit/core/TestInstallAddon.js]: Yeah every one is definitely excessive, I was just thinking at least one "Unix". 😄

File: src/core/installAddon.js (hunk 1 above)
Comment (willdurand) [reply src/core/installAddon.js]: 😈

**Pull request merged**